File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
         var maxWater = 0;
	for (int left = 0, right = height.length - 1, distance = right; distance > 0; distance--)
		maxWater = Math.max(maxWater, distance * (height[left] < height[right] ? 
												  height[left++] : 
												  height[right--]));
	return maxWater;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
         int maximum = 0, current;
	for(int i = 0, j=n-1, k=j; k>0; k--){
        
    current = (height[i] < height[j] ?  height[i++] : height[j--])*k;  
	maximum = max(maximum, current);
        
    }
        
	return maximum;
    }   
};